### PR TITLE
Gene identifier/annotation fixes

### PR DESCRIPTION
**Pull request by SJShaw**
Removes all `No gene ID` identifiers, replacing them with valid identifiers if possible or removing the fields if not.

Removes or corrects all duplications of genes in gene annotations, as the identifier should be unique, The validator script should now prevent new cases of this appearing.

Corrected, and forbidden in the schema for future cases, any gene identifiers containing trailing or leading whitespace.

Fixes a batch of entries where the polyketide module gene list was a single string containing comma separated gene names instead of being an actual list of strings. The schema has changed to forbid commas in any gene names in that list. Commas remain in the gene annotations `name` field for some entries, but will be handled later on.

### CODE REVIEW

File: schema.json
Comment (kblin) [inline schema.json:1168]: Whitespace change

**Review by kblin — CHANGES_REQUESTED** [review]
Apart from the extra newline in the schema, 👍 from me.

**Comment by SJShaw** [comment]
I've a branch ready that rebases this on top of #193, so it should wait until that's merged.

**Comment by kblin** [comment]
Ok, #193 is merged, I'm happy to look over the rebased branch again.

**Review by kblin — APPROVED** [review]
Looks good to me 👍

**Pull request merged**
